Automattic/newspack-ios
# Task/housekeeping

This PR addresses some miscellaneous housekeeping tasks that don't really belong in a feature PR. 

- A flaky unit test is fixed for good (I think 😬 )
- Model defaultFetchRequests are refactored to a protocol extension (h/t @jleandroperez for the idea and approach!) 
- Folder name sanitization is moved into the FolderManager. This seems a better place than an extension.
- Bookmark helpers are also moved into the FolderManager. I've opted for this location mostly due to the need to detect trashed assets--something that should be handled by a FileManager (if its broken API is ever fixed) and the FolderManager is intended to encapsulate FileManager access.
- The FolderKeeper protocol/extension is nuked.
- A method to check if a URL points to a trashed resource is added.
- The Site model's siteFolder property is no longer optional.

Unit test are updated according to the changes above.

To test:
- Run unit tests. Ensure everything passes. 
- Launch the app and make a cursory inspection of the Story folder features.

@jleandroperez can I trouble you for a review?
